internal/formats/common/spdxhelpers/to_syft_model.go

```diff
@@ -0,0 +1,373 @@
+								IDLike:     []string{name},
+								Version:    version,
+								VersionID:  version,
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func findSyftLinuxRelease(doc *spdx.Document2_2) *linux.Release {
+	r := findLinuxReleaseByDocument(doc)
+	if r != nil {
+		return r
+	}
+
+	r = findLinuxReleaseByPURL(doc)
+	if r != nil {
+		return r
+	}
+
+	return nil
+}
+
+func findSpdxRelationshipByType(doc *spdx.Document2_2, typ string) *spdx.Relationship2_2 {
+	for _, r := range doc.Relationships {
+		if typ == r.Relationship {
+			return r
+		}
+	}
+	return nil
+}
+
+func findSpdxPackageByID(doc *spdx.Document2_2, id spdx.ElementID) *spdx.Package2_2 {
+	for _, p := range doc.Packages {
+		if p.PackageSPDXIdentifier == id {
+			return p
+		}
+	}
+	return nil
+}
+
+func findSpdxReferenceByName(p *spdx.Package2_2, categories ...string) *spdx.PackageExternalReference2_2 {
+	for _, r := range p.PackageExternalReferences {
+		for _, category := range categories {
+			if r.Category == category {
+				return r
+			}
+		}
+	}
+	return nil
+}
+
+func collectSyftFiles(spdxIDMap map[string]interface{}, doc *spdx.Document2_2) {
+	for _, f := range doc.UnpackagedFiles {
+		spdxIDMap[string(f.FileSPDXIdentifier)] = toSyftLocation(f)
+	}
+}
+
+func toSyftRelationships(spdxIDMap map[string]interface{}, doc *spdx.Document2_2) []artifact.Relationship {
+	var out []artifact.Relationship
+	for _, r := range doc.Relationships {
+		// FIXME what to do with r.RefA.DocumentRefID and  r.RefA.SpecialID
+		if r.RefA.DocumentRefID != "" && requireAndTrimPrefix(r.RefA.DocumentRefID, "DocumentRef-") != string(doc.CreationInfo.SPDXIdentifier) {
+			log.Debugf("relationship to external document: %+v", r)
+			continue
+		}
+		a := spdxIDMap[string(r.RefA.ElementRefID)]
+		b := spdxIDMap[string(r.RefB.ElementRefID)]
+		from, fromOk := a.(*pkg.Package)
+		toPackage, toPackageOk := b.(*pkg.Package)
+		toLocation, toLocationOk := b.(*source.Location)
+		if !fromOk || !(toPackageOk || toLocationOk) {
+			log.Debugf("unable to find valid relationship mapping from SPDX 2.2 JSON, ignoring: (from: %+v) (to: %+v)", a, b)
+			continue
+		}
+		var to artifact.Identifiable
+		var typ artifact.RelationshipType
+		switch r.Relationship {
+		case "CONTAINS":
+			if toLocationOk {
+				from.Locations = append(from.Locations, *toLocation)
+				typ = artifact.ContainsRelationship
+				to = toLocation
+			}
+		case "OVERWRITE":
+			typ = artifact.OwnershipByFileOverlapRelationship
+			to = toPackage
+		}
+		if typ != "" && to != nil {
+			out = append(out, artifact.Relationship{
+				From: from,
+				To:   to,
+				Type: typ,
+				Data: nil, // FIXME should there be anything for this Data?
+			})
+		}
+	}
+	return out
+}
+
+func collectSyftPackages(spdxIDMap map[string]interface{}, doc *spdx.Document2_2) {
+	for _, p := range doc.Packages {
+		syftPkg := toSyftPackage(p)
+		spdxIDMap[string(p.PackageSPDXIdentifier)] = syftPkg
+		for _, f := range p.Files {
```

Comment:
it looks like we're decoding `package.Files` into the syft `pkg.Package.Locations` however the inverse is not true (we are not encoding `pkg.Package.Locations` into `package.Files` when making SPDX documents). As a high level comment we should be in parity in the encoding and decoding paths for how we interpret/use fields.

This comment may apply to other fields as well.

Reply:
I'm not sure that is entirely accurate. We should try our best to process any incoming SPDX document rather than just support those created by Syft.